### PR TITLE
Add repo and workflow dispatches (SOFTWARE-4651)

### PR DESCRIPTION
**Pull request by brianhlin**

### CODE REVIEW

File: .github/workflows/metrics.yml [hunk 1]
```diff
@@ -5,6 +5,11 @@ on:
     branches:
       - master
 
+  repository_dispatch:
+    types: [dispatch-metrics]
+
+  workflow_dispatch:
```
Comment (edquist) [inline .github/workflows/metrics.yml:11]: does `workflow_dispatch` work with no value?

Comment (brianhlin) [reply .github/workflows/metrics.yml:11]: Yup! This is apparently legal YAML https://github.com/opensciencegrid/docker-software-base/blob/master/.github/workflows/build-container.yml#L11

See the nifty `Run workflow` button that it adds here: https://github.com/opensciencegrid/docker-software-base/actions/workflows/build-container.yml

Comment (edquist) [reply .github/workflows/metrics.yml:11]: So much for me believing that yaml was a respectable language  ;)

**Review by edquist — APPROVED** [review]
Go go gadget run-buttons!

**Pull request merged**